pkg/lib/aws/s3.go

```diff
@@ -334,144 +428,218 @@ func (c *Client) ReadBytesFromS3Path(s3Path string) ([]byte, error) {
 	return c.ReadBytesFromS3(bucket, key)
 }
 
-func (c *Client) ReadBytesFromS3(bucket string, key string) ([]byte, error) {
-	response, err := c.S3().GetObject(&s3.GetObjectInput{
-		Key:    aws.String(key),
+// overwrites existing file
+func (c *Client) DownloadFileFromS3(bucket string, key string, localPath string) error {
+	file, err := files.Create(localPath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	// for downloading files, s3manager.Downloader.Download() is faster than s3.S3.GetObject()
+	_, err = c.S3Downloader().Download(file, &s3.GetObjectInput{
 		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
 	})
 
 	if err != nil {
-		return nil, errors.Wrap(err, S3Path(bucket, key))
+		return errors.Wrap(err, S3Path(bucket, key))
 	}
 
-	buf := new(bytes.Buffer)
-	buf.ReadFrom(response.Body)
-	return buf.Bytes(), nil
+	return nil
 }
 
-func (c *Client) ListDir(bucket string, prefix string, maxResults int64) ([]*s3.Object, error) {
-	prefix = s.EnsureSuffix(prefix, "/")
-	return c.ListPrefix(bucket, prefix, maxResults)
+func (c *Client) DownloadDirFromS3(bucket string, s3Dir string, localDirPath string, shouldTrimDirPrefix bool, maxFiles *int64) error {
+	prefix := s.EnsureSuffix(s3Dir, "/")
+	return c.DownloadPrefixFromS3(bucket, prefix, localDirPath, shouldTrimDirPrefix, maxFiles)
 }
 
-func (c *Client) ListPathDir(s3Path string, maxResults int64) ([]*s3.Object, error) {
-	s3Path = s.EnsureSuffix(s3Path, "/")
-	return c.ListPathPrefix(s3Path, maxResults)
-}
+// if shouldTrimDirPrefix is true, the directory path of prefix will be trimmed when downloading files
+//   e.g. if prefix = "test/dir", "test/" will be trimmed when copying; if prefix = "test/dir/", "test/dir/" will be trimmed
```

Comment:
I think this comment can be improved with an example directory structure in S3 and show how the files will be copied to locally. I think I understand what this is saying but having an example will give me more confidence that I will use this function correctly.